hkdf/src/hkdf.rs

```diff
@@ -27,19 +27,26 @@ impl<D> Hkdf<D>
           D::OutputSize: ArrayLength<u8>
 {
     #[deprecated(since="0.4.0", note="renamed to extract, with arguments in reversed order")]
-    pub fn new(ikm: &[u8], salt: &[u8]) -> Hkdf<D> {
+    pub fn new(ikm: &[u8], salt: Option<&[u8]>) -> Hkdf<D> {
         Self::extract(salt, ikm)
     }
```

Comment:
Should we change signature of the `new` as well? I think we either should remove it altogether (the `extract` change will require minor version bump anyway) or undeprecate it and and leave signature as `pub fn new(ikm: &[u8], salt: &[u8])`.

Reply:
Because this only exists for compatibility, it should either remain as-is or be removed. Promoting it would add unnecessary confusion vs. `extract`.